engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/TimeOfDayBranchNode.java

```diff
@@ -0,0 +1,59 @@
+package org.enso.interpreter.node.controlflow.caseexpr;
+
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import org.enso.interpreter.runtime.data.EnsoTimeOfDay;
+import org.enso.interpreter.runtime.data.Type;
+
+/** An implementation of the case expression specialised to working on Time_Of_Day. */
+@NodeInfo(shortName = "TimeOfDayMatch")
+public abstract class TimeOfDayBranchNode extends BranchNode {
```

Comment:
Sadly that matching won't work atm without BranchNodes. I wanted to avoid writing them because, yes, they will go away soon, but with the current approach they are still necessary.